src/debug-server.ts

```diff
@@ -20,10 +21,48 @@ export interface DebugCommand {
 
 export interface DebugStep {
     type: 'setBreakpoint' | 'removeBreakpoint' | 'continue' | 'evaluate' | 'launch';
-    file: string;
+    file?: string;
     line?: number;
     expression?: string;
     condition?: string;
+    timeoutMs?: number;
+}
+
+// New structured response types
+export interface DebugStepResult {
```

Comment:
which parts of this system should actually be modeled as a resources vs tools? https://modelcontextprotocol.io/specification/2025-06-18/server/resources